### DISCUSSION TITLE
Backup of PVC using a local disk to save the backup

### DISCUSSION THREAD

**Discussion by jmsecconi** [vmware-tanzu/velero#5645] [Community support Q&A]
I run K8S on a private network with no accès to the internet, so I can't use any cloud provider's plugin. 
I read about using File System backup using restic, but it is not clear how to specify in velero that I want to use the /data/backup/k8s directory of my local system in velero.

Can you precise it?

**Comment by sseago** [comment]
Velero's Restic operation stores the file system backup in the BackupStorageLocation (BSL)  that the rest of the backup is stored in. Also, the restic filesystem backup is *part* of a velero backup, it doesn't stand alone, so you still need the BSL to store the kubernetes resoure metadata (including the PV/PVC metadata needed for volume restore along with the restic snapshot). One thing you can do is set up a local s3-compatible object store in your cluster (such as Minio), and then create a backup storage location pointing to your minio s3 endpoint. There have been some discussions about possibly writing an ObjectStore plugin backed by a filesystem (such as an NFS share) -- if that were to be created, then you'd be able to use a regular NFS share to store your backups, but nothing like that exists in Velero today.

**Reply by jmsecconi** [reply]
This means that if I save my backup in a minio instance, the backup will be inside my cluster.
What I need is to have the backup outside of my cluster to be able to save it and restore it with regular backup tools
Do you mean it is not possible with velero in an isolated network?

**Reply by sseago** [reply]
You could also create an s3-compatible object store outside of your cluster but inside your isolated network. Velero requires a BackupStorageLocation in which to store a backup. The velero ObjectStore API is pluggable, but the only currently-supported plugin which will work in a non-cloud environment is the s3/aws plugin (since it's possible to have an s3-compatible object store on an isolated network). What we don't have right now is an ObjectStore plugin which works on a bare filesystem mount back end. That would be needed if you wanted to create Velero backups without an s3 store inside your isolated network.
